src/contracts/fund/trading/calls/getExchangeIndex.ts

```diff
@@ -0,0 +1,29 @@
+import { getContract } from '~/utils/solidity';
+import { Contracts } from '~/Contracts';
+import { ensure } from '~/utils/guards/ensure';
+
+const getExchangeIndex = async (
```

Comment:
The general function signature is:
```ts
const fn = (contractAddress, paramsObject, environment)
```

Where as the `contractAddress` is always the address of the contract this function acts upon and `paramsObject` should always be an object.

So this signature should be:
```ts
const getExchangeIndex = async (tradingAddress, { exchangeAddress }, environment)
```

Reply:
Usage of the `callFactory` would enforce this behaviour and also give added niceties like the possibility to subscribe to a call.